app/controllers/application_controller/indexes.rb

```diff
@@ -0,0 +1,380 @@
+  #
+  ##############################################################################
+
+  # Dispatch to a subaction
+  def index
+    self.class.index_subaction_param_keys.each do |subaction|
+      if params[subaction].present?
+        return send(
+          self.class.index_subaction_dispatch_table[subaction] ||
+          subaction
+        )
+      end
+    end
+    default_index_subaction
+  end
+
+  # Render an index or set of search results as a list or matrix. Arguments:
+  # query::     Query instance describing search/index.
+  # args::      Hash of options.
+  #
+  # Options include these:
+  # id::            Warp to page that includes object with this id.
+  # action::        Template used to render results.
+  # matrix::        Displaying results as matrix?
+  # cache::         Cache the HTML of the results?
+  # letters::       Paginating by letter?
+  # letter_arg::    Param used to store letter for pagination.
+  # number_arg::    Param used to store page number for pagination.
+  # num_per_page::  Number of results per page.
+  # always_index::  Always show index, even if only one result.
+  #
+  # Side-effects: (sets/uses the following instance variables for the view)
+  # @title::        Provides default title.
+  # @layout::
+  # @pages::        Paginator instance.
+  # @objects::      Array of objects to be shown.
+  # @extra_data::   Results of block yielded on every object if block given.
+  #
+  # Other side-effects:
+  # store_location::          Sets this as the +redirect_back_or_default+
+  #                           location.
+  # clear_query_in_session::  Clears the query from the "clipboard"
+  #                           (if you didn't just store this query on it!).
+  # query_params_set::        Tells +query_params+ to pass this query on
+  #                           in links on this page.
+  #
+  def show_index_of_objects(query, args = {})
+    show_index_setup(query, args)
+    if (@num_results == 1) && !args[:always_index]
+      show_action_redirect(query)
+    else
+      calc_pages_and_objects(query, args)
+      if block_given?
+        @extra_data = @objects.each_with_object({}) do |object, data|
+          row = yield(object)
+          row = [row] unless row.is_a?(Array)
+          data[object.id] = row
+        end
+      end
+      show_index_render(args)
+    end
+  end
+
+  private ##########
+
+  def show_index_setup(query, args)
+    apply_content_filters(query)
+    store_location
+    clear_query_in_session if session[:checklist_source] != query.id
+    query_params_set(query)
+    query.need_letters = args[:letters] if args[:letters]
+    set_index_view_ivars(query, args)
+  end
+
+  # rubocop:disable Rails/FindEach
+  def apply_content_filters(query)
+    filters = users_content_filters || {}
+    @any_content_filters_applied = false
+    ContentFilter.all.each do |fltr|
+      apply_one_content_filter(fltr, query, filters[fltr.sym])
+    end
+  end
+  # rubocop:enable Rails/FindEach
+
```

Comment:
1. @mo-nathan had long ago suggested leaving a comment  explaining why a cop is disabled.
2. No need to disable the cop for the whole method.
3. If just one line is affected, I like to disable cops with a an in-line comment, which results in two fewer lines.
```suggestion
  def apply_content_filters(query)
    filters = users_content_filters || {}
    @any_content_filters_applied = false
    # disable cop because ContentFilter is not an ActiveRecord model
    ContentFilter.all.each do |fltr| # rubocop:disable Rails/FindEach
      apply_one_content_filter(fltr, query, filters[fltr.sym])
    end
  end
```